src/benchmarks/micro/libraries/System.Utf8String/Perf.Utf8String.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.IO;
+
+namespace System.Text
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.Runtime)]
+    public class Perf_Utf8String
+    {
+        // the benchmark uses text files from Project Gutenberg
+        public enum InputFile
+        {
+            EnglishAllAscii, // English, all-ASCII, should stay entirely within fast paths
+            EnglishMostlyAscii, // English, mostly ASCII with some rare non-ASCII chars, exercises that the occasional non-ASCII char doesn't kill our fast paths
+            Chinese, // Chinese, exercises 3-byte scalar processing paths typical of East Asian languages
+            Cyrillic, // Cyrillic, exercises a combination of ASCII and 2-byte scalar processing paths
+            Greek, // Greek, similar to the Cyrillic case but with a different distribution of ASCII and non-ASCII chars
+        }
+
+        [ParamsAllValues] // BDN uses all values of given enum
+        public InputFile Input { get; set; }
+
+        private Utf8String _utf8;
+        private Memory<char> _destination;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            string unicode = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "libraries", "System.Utf8String", $"{Input}.txt"));
+            _utf8 = new Utf8String(unicode);
+            _destination = new char[unicode.Length];
+        }
+
+        [Benchmark]
+        public int ToChars() => new Utf8Span(_utf8).ToChars(_destination.Span);
```

Comment:
@adamsitnik  - Should we avoid creating `Utf8Span()` inside this method but instead have it created globally?

Reply:
@kunalspathak this is what we do by default, but `Utf8Span` is a stack-only struct so we can't store it in the class field and BDN does not support `ref structs` as type that define benchmarks

https://github.com/dotnet/runtime/blob/b5705587347d29d79cec830dc22b389e1ad9a9e0/src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs#L20